docs/config/index.md

```diff
@@ -134,6 +134,10 @@ export default defineConfig(({ command, mode }) => {
   - Full URL, e.g. `https://foo.com/`
   - Empty string or `./` (for embedded deployment)
 
+  Beware that using a relative `base` path for a single-page web-based app may break when calls to window.history.pushState change the path in the address bar, as your code will attempt to load assets relative to that modified path in production, but often not in serve (AKA dev) mode used for development testing. This can cause workers to silently fail to load in production when the relative path fails to find the worker that appeared to be at the correct location at compile time and when running via serve/dev.
+  
+  Beware that using an absolute `base` path will likely break electron and other apps that run on the local filesystem and are not served via the web, as these apps may be installed in different locations and should thus rely on file-system based relative paths. For example, when building such filesystem-based app, a `base` of `""` (or equivalently `"./"`) will cause `vite` to use paths that will be relative to the current directory when the code is executed.
+
```

Comment:
I think we should put these under the `See Public Base Path` paragraph. And make a "tip" blockquote like:

```
::: tip Relative base path

<description>

:::
```

I'd try to shorten the description a bit too as a long paragraph increases the [cognitive overhead](https://github.com/vuejs/v2.vuejs.org/blob/master/writing-guide.md#principles).

I'm not sure if we need the second paragraph, since Vite doesn't have official support for Electron. Perhaps it can be in https://github.com/cawa-93/vite-electron-builder. But I'm open if someone else if fine having it here.

Reply:
Thanks.

Regardless of whether electron itself is specifically supported, vite will surely be used to build apps that are run out of the file system.  So, while the second paragraph need not be as hard to ignore as the first, it would be valuable to have it be found by a search within the official vite documentation.

The danger of relative paths should be made hard to ignore because vite won't do anything to detect the problem, the dev server will hide the problem, and it's very hard to identify when debugging a production website.